### PR TITLE
ENH: Add vtkITKGrowCut filter

### PR DESCRIPTION
**Pull request by Sunderlandkyl**
This commit adds a new VTK filter vtkITKGrowCut, which uses the itk::FastGrowCut implementation.
The new filter is used in the "Grow from seeds" segment editor effect.
SegmentationsModuleTest2 now includes a Grow from seeds test.

### CODE REVIEW

**Review by lassoan — COMMENTED** [review]
Thank you @Sunderlandkyl this works very well. I've found only one small issue that would need to be fixed before integrating: seed locality does not have an effect unless you cancel and re-initialize.

Example:

Default:

image

Setting Seed locality to 10 has no effect:

image

After Cancel and Initialize it provides correct result:

image

It would probably need to be fixed in ITK (or at least file a bug report in ITK) to invalidate the internal computed distances if the distance penalty value is modified.

But as an immediate workaround in Slicer (to enable merging without waiting for the fix in ITK) could be to cancel and initialize the filter.

**Comment by lassoan** [comment]
@Sunderlandkyl the number of open issues for the new stable release (https://github.com/Slicer/Slicer/milestone/6) is decreasing quite rapidly, so it seems that we can cut the new release soon. It would be nice if you could fix the small seed locality update issue (see above) by early next week because then this PR could be included in this release. Thanks in advance!

**Comment by Sunderlandkyl** [comment]
Thanks for the review Andras!
I've added a Reset() call when the distance penalty is changed.

File: Libs/vtkITK/vtkITKGrowCut.cxx [hunk 1]
```diff
@@ -0,0 +1,259 @@
...
+}
+
+//-----------------------------------------------------------------------------
+void vtkITKGrowCut::SetDistancePenalty(double distancePenalty)
+{
+  vtkDebugMacro(<< this->GetClassName() << " (" << this << "): setting DistancePenalty to " << distancePenalty);
+  if (this->DistancePenalty != distancePenalty)
+    {
+    this->DistancePenalty = distancePenalty;
+    // TODO: Due to a bug in ITK, we currently need to reset the filter when changing the distance penalty.
```
Comment (lassoan) [inline Libs/vtkITK/vtkITKGrowCut.cxx:254]:
```suggestion
    // TODO: Due to a bug in ITK (https://github.com/InsightSoftwareConsortium/ITKGrowCut/issues/18),
    // we currently need to reset the filter when changing the distance penalty.
```

**Review by lassoan — APPROVED** [review]
Thank you! I've added an ITK bug report (https://github.com/InsightSoftwareConsortium/ITKGrowCut/issues/18)

**Pull request merged**